Chess/knight.cpp

```diff
@@ -0,0 +1,35 @@
+#include "knight.h"
+
+
+Knight::Knight(int row, int column) : Piece(row, column) {
+
+}
+
+
+void Knight::makeMove(int diffRow, int diffColumn) {
+
+    vector <MoveBy> moves = getMoves();
+
+    bool ok = 0;
+
+    MoveBy currentMove = {diffRow, diffColumn};
+
+    for(auto i : moves) {
+        ok |= (i == currentMove && this->isValidMove(diffRow, diffColumn));

```

Comment:
1.
`this->isValidMove(diffRow, diffColumn);`

Nu depinde de i, poate fi facuta o singura data la inceput

2. Pare ca asta e mai mult o validare care verifica daca mutarea e legala (regulamentara).
Poti sa o scoti intr-o functie separata de `isLegalMove(diffRow, diffColumn)` care sa intoarca true sau false daca este in acel vector de `moves`. Vezi exemplu la clasa `Bishop`